### PR TITLE
[FUSE-68] - BE: Implement Style for Trainer (Many-to-Many)

### PR DESCRIPTION
**Pull request by CaoSon1909**
- Implemented Trainer Style (M-M)
- Get all trainers:
 + 
image
+
image
+
image
- View details of the trainer:
+ 
image
+
image
+
image

### CODE REVIEW

File: apps/weight-loss-consultant-users-mgnt-api/src/app/services/sorting-filtering.service.ts [hunk 1]
```diff
@@ -105,14 +107,16 @@ export class SortingAndFilteringService {
...
+          .where("gender = :gender", {gender: genderFilter === "1" ? Gender.MALE : Gender.FEMALE})
+          .leftJoinAndSelect("customer.campaigns", "campaign")
           .orderBy(sortBy, order === "ASC" ? "ASC" : "DESC")
           .offset(skipped)
           .limit(limit)
           .getMany()
-        const totalCount = await this.customerRepository.createQueryBuilder()
-          .where("gender = :gender", {gender: genderFilter === "Male" ? "Male" : "Female"})
+        const totalCount = await this.customerRepository.createQueryBuilder("customer")
+          .where("customer.gender = :gender", {gender: genderFilter === "1" ? Gender.MALE : Gender.FEMALE})
```
Comment (TienTruong2000) [inline apps/weight-loss-consultant-users-mgnt-api/src/app/services/sorting-filtering.service.ts:118]: What is "1" here means? Should separate to another variable or add comment.

File: apps/weight-loss-consultant-users-mgnt-api/src/app/services/sorting-filtering.service.ts [hunk 2]
```diff
@@ -75,14 +74,16 @@ export class SortingAndFilteringService {
...
+          .where("trainer.gender = :gender", {gender: genderFilter === "1" ? Gender.MALE : Gender.FEMALE})
           .orderBy(sortBy, order === "ASC" ? "ASC" : "DESC")
           .offset(skipped)
           .limit(limit)
           .getMany()
-        const totalCount = await this.trainerRepository.createQueryBuilder()
-          .where("gender = :gender", {gender: genderFilter === "Male" ? "Male" : "Female"})
+        const totalCount = await this.trainerRepository.createQueryBuilder("trainer")
+          .where("trainer.gender = :gender", {gender: genderFilter === "1" ? Gender.MALE : Gender.FEMALE})
+          .leftJoinAndSelect("trainer.packages", "package")
```
Comment (TienTruong2000) [inline apps/weight-loss-consultant-users-mgnt-api/src/app/services/sorting-filtering.service.ts:86]: What is "1" here means? Should separate to another variable or add comment.

File: apps/weight-loss-consultant-users-mgnt-api/src/app/services/sorting-filtering.service.ts [hunk 3]
```diff
@@ -75,14 +74,16 @@ export class SortingAndFilteringService {
         break;
       }
       default: {
-        const result = await this.trainerRepository.createQueryBuilder()
-          .where("gender = :gender", {gender: genderFilter === "Male" ? "Male" : "Female"})
+        const result = await this.trainerRepository.createQueryBuilder("trainer")
+          .leftJoinAndSelect("trainer.packages", "package")
+          .where("trainer.gender = :gender", {gender: genderFilter === "1" ? Gender.MALE : Gender.FEMALE})
           .orderBy(sortBy, order === "ASC" ? "ASC" : "DESC")
```
Comment (TienTruong2000) [inline apps/weight-loss-consultant-users-mgnt-api/src/app/services/sorting-filtering.service.ts:80]: What is "1" here means? Should separate to another variable or add comment.

Comment (CaoSon1909) [reply apps/weight-loss-consultant-users-mgnt-api/src/app/services/sorting-filtering.service.ts]: ok I will fix it

File: apps/weight-loss-consultant-users-mgnt-api/src/app/services/sorting-filtering.service.ts (hunk 2 above)
Comment (CaoSon1909) [reply apps/weight-loss-consultant-users-mgnt-api/src/app/services/sorting-filtering.service.ts]: ok I will fix it

File: apps/weight-loss-consultant-users-mgnt-api/src/app/services/sorting-filtering.service.ts (hunk 1 above)
Comment (CaoSon1909) [reply apps/weight-loss-consultant-users-mgnt-api/src/app/services/sorting-filtering.service.ts]: ok I will fix it

**Pull request merged**